xen/arch/arm/tee/ffa.c

```diff
@@ -502,36 +494,48 @@ static bool ffa_probe(void)
     {
         printk(XENLOG_ERR "ffa: Incompatible version %#x found\n", vers);
     }
-    /*
-     * At the moment domains must support the same features used by Xen.
-     * TODO: Rework the code to allow domain to use a subset of the
-     * features supported.
-     */
-    else if ( !check_mandatory_feature(FFA_PARTITION_INFO_GET) ||
-         !check_mandatory_feature(FFA_RX_RELEASE) ||
-         !check_mandatory_feature(FFA_RXTX_MAP_64) ||
-         !check_mandatory_feature(FFA_MEM_SHARE_64) ||
-         !check_mandatory_feature(FFA_RXTX_UNMAP) ||
-         !check_mandatory_feature(FFA_MEM_SHARE_32) ||
-         !check_mandatory_feature(FFA_MEM_RECLAIM) ||
-         !check_mandatory_feature(FFA_MSG_SEND_DIRECT_REQ_32) )
-    {
-        printk(XENLOG_ERR "ffa: Mandatory feature not supported by fw\n");
-    }
     else
     {
+        uint32_t featnum;
+
         major_vers = (vers >> FFA_VERSION_MAJOR_SHIFT)
                      & FFA_VERSION_MAJOR_MASK;
         minor_vers = vers & FFA_VERSION_MINOR_MASK;
         printk(XENLOG_INFO "ARM FF-A Firmware version %u.%u\n",
                major_vers, minor_vers);
 
         ffa_fw_version = vers;
+
+        for (featnum = 0; featnum < 64; featnum++)
```

Comment:
Would it make sense to use a define for the maximum feature number?

Reply:
maybe something using sizeof(ffa_fw_feat32/64_supported) would make more sense ?

Reply:
How about using a proper bitmap instead, manipulated with `clear_bit()`, `set_bit()`, and so on?
That way you define what capacity you need and that's it. I doubt there would be any overhead.

Reply:
i think you are right, i will do that